awx/sso/backends.py

```diff
@@ -395,10 +395,17 @@ def on_populate_user(sender, **kwargs):
         remove = bool(team_opts.get('remove', True))
         _update_m2m_from_groups(user, ldap_user, team.member_role.members, users_opts, remove)
 
+    # Check if user.profile is available, otherwise force user.save()
+    try:
+        _ = user.profile
+    except ValueError:
+        force_user_update = True
+    finally:
+        if force_user_update:
```

Comment:
```>>> try:
...   raise Exception
... except ValueError:
...   print("hello")
... finally:
...   print("good bye")
...
good bye
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
Exception
```

You can end up in the `finally` due to some exception other than `ValueError` & `force_user_update` will be an undefined variable.

TL;DR set `force_user_update = False` before the `try:`

Reply:
@chrismeyersfsu the `force_user_update = False` is already set before the `try:`  statement at https://github.com/ansible/awx/blob/devel/awx/sso/backends.py#L357